pyam/core.py

```diff
@@ -806,9 +806,11 @@ def set_meta(self, meta, name=None, index=None):
                 meta = meta.rename(
                     columns={i.capitalize(): i for i in META_IDX}
                 ).set_index(self.meta.index.names)
+
             meta = meta.loc[self.meta.index.intersection(meta.index)]
+            meta.index = meta.index.remove_unused_levels()
             self.meta = merge_meta(meta, self.meta, ignore_conflict=True)
-            return
+            return  # EXIT FUNCTION
```

Comment:
Who/what is this comment for the benefit of?

Reply:
It's been used in a few places nearby, so I added it for consistency - per your comment, I removed it everywhere...